command-utility-tool/src/main/java/com/utility/FileScanner.java

```diff
@@ -0,0 +1,108 @@
+package com.utility;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.*;
+
+/**
+ * This class scans the input file and stores word-occurrence pairs
+ */
+public class FileScanner {
+
+//        1. - use map to store
+//        1. key:  word all lowercase
+//        2. value: occurrence
+    Map<String, List<String>> wordByOccurrence = new HashMap<>();
+
+    /**
+     * Scan the input file and save with word-occurrence pairs
+     * @param filePath
+     */
+    public void scanFile(String filePath) {
+        try {
+            File file = new File(filePath);
+            Scanner scan = new Scanner(file);
+            //Reading each line of the file using Scanner class
+            int lineNumber = 1;
+            while(scan.hasNextLine()){
+                String line = scan.nextLine().toLowerCase();
+
+                List<String> lineWords = getLineWords(line);
+                System.out.println(Arrays.toString(lineWords.toArray()));
+//                System.out.println(Arrays.toString(lineWords));
+                for(String word: lineWords) {
+                    System.out.println(word);
+
+                    // remove any special characters at the end
+                    for (int i = 0; i < line.length() - word.length() + 1 ; i++) {
+                        int column = -1;
+                        String currentStr = line.substring(i, i + word.length());
+                        System.out.println(currentStr);
+
+                        if (currentStr.equals(word)) {
+                            column = i;
+                            String occurrence = "Line: " + lineNumber + " Column: " + column;
```

Comment:
This will work on a simple application like this. So the `wordByOccurrence` map contains a pair of a word and a list of the static strings, i.e. `"Line: " + lineNumber + " Column: " + column`.

If we think of the usages, the client will only be able to retrieve the formatted string. In other words, the client cannot get an independent `lineNumber` or `column`.  So what if the client wants to retrieve their values and create their own String like "ln: 3, column: 7" instead of "Line: 3  Column: 7". 

So in this case, you can create another type, e.g. `class Location{ // }`, and have private instance variables of the lineNumber and the column. Then the client can handle them however they like.

Reply:
good point, will do